src/components/SDFlatButton.react.js

```diff
@@ -0,0 +1,222 @@
+  label?: string,
+  labelPosition?: string,
+  labelStyle?: Object,
+  n_clicks?: number,
+  onClick?: () => void,
+  onKeyboardFocus?: () => void,
+  primary?: boolean,
+  rippleColor?: string,
+  secondary?: boolean,
+  setProps?: () => void,
+  style?: Object,
+};
+
+const propTypes = {
+  /**
+   * Color of button when mouse is not hovering over it.
+   */
+  backgroundColor: PropTypes.string,
+
+  /**
+   * This is what will be displayed inside the button.
+   * If a label is specified, the text within the label prop will
+   * be displayed. Otherwise, the component will expect children
+   * which will then be displayed. (In our example,
+   * we are nesting an `<input type="file" />` and a `span`
+   * that acts as our label to be displayed.) This only
+   * applies to flat and raised buttons.
+   */
+  children: PropTypes.node,
+
+  /**
+   * The CSS class name of the root element.
+   */
+  className: PropTypes.string,
+
+  /**
+   * The element to use as the container for the FlatButton. Either a string to
+   * use a DOM element or a ReactElement. This is useful for wrapping the
+   * FlatButton in a custom Link component. If a ReactElement is given, ensure
+   * that it passes all of its given props through to the underlying DOM
+   * element and renders its children prop for proper integration.
+   */
+  containerElement: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+
+  /**
+   * If true, the element's ripple effect will be disabled.
+   */
+  disableTouchRipple: PropTypes.bool,
+
+  /**
+   * Disables the button if set to true.
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * If true, the button will take up the full width of its container.
+   */
+  fullWidth: PropTypes.bool,
+
+  /**
+   * Color of button when mouse hovers over.
+   */
+  hoverColor: PropTypes.string,
+
+  /**
+   * The URL to link to when the button is clicked.
+   */
+  href: PropTypes.string,
+
+  /**
+   * Use this property to display an icon.
+   */
+  icon: PropTypes.node,
+
+  /* The element's ID */
+  id: PropTypes.string,
+
+  /**
+   * Label for the button.
+   */
+  label: PropTypes.string,
+
+  /**
+   * Place label before or after the passed children.
+   */
+  labelPosition: PropTypes.oneOf(['before', 'after']),
+
+  /**
+   * Override the inline-styles of the button's label element.
+   */
+  labelStyle: PropTypes.object,
+
+  /**
+   * An integer that represents the number of times
+   * that this element has been clicked on.
+   */
+  n_clicks: PropTypes.integer,
+
+  /**
+   * Callback function fired when the button is clicked.
+   *
+   * @param {object} event Click event targeting the button.
+   */
+
+  onClick: PropTypes.func,
+
+  /**
+   * Callback function fired when the element is focused or blurred by the keyboard.
+   *
+   * @param {object} event `focus` or `blur` event targeting the element.
+   * @param {boolean} isKeyboardFocused Indicates whether the element is focused.
+   */
+  onKeyboardFocus: PropTypes.func,
```

Comment:
What does including this do? Is there any way to call this from Dash?

Reply:
I do not think it can be called from Dash; it was not listed as an available property. I included it to try to make the FlatButton port as 1-to-1 as possible, but it isn't necessary.